src/library/coverartcache.cpp

```diff
@@ -114,16 +105,27 @@ QPixmap CoverArtCache::requestCover(const CoverInfo& requestInfo,
     QFutureWatcher<FutureResult>* watcher = new QFutureWatcher<FutureResult>(this);
     QFuture<FutureResult> future = QtConcurrent::run(
             this, &CoverArtCache::loadCover, requestInfo, pRequestor,
-            requestReference, desiredWidth, signalWhenDone);
+            desiredWidth, signalWhenDone);
     connect(watcher, SIGNAL(finished()), this, SLOT(coverLoaded()));
     watcher->setFuture(future);
     return QPixmap();
 }
 
+//static
+void CoverArtCache::requestCover(const Track* pTrack,
+                         const QObject* pRequestor) {
+    CoverArtCache* pCache = CoverArtCache::instance();
+    if (pCache == nullptr || pTrack == nullptr) return;
+
+    CoverInfo info = pTrack->getCoverInfo();
+    // trackLocation is still empty here
```

Comment:
Document and verify your assumption with a debug assertion instead of a comment:
DEBUG_ASSERT(info.trackLocation.isEmpty());